server/pulp/server/db/querysets.py

```diff
@@ -1,10 +1,44 @@
-from mongoengine.queryset import DoesNotExist, QuerySet
+from mongoengine.queryset import DoesNotExist, QuerySetNoCache
 from pymongo import ASCENDING
 
 from pulp.server import exceptions as pulp_exceptions
 
 
-class CriteriaQuerySet(QuerySet):
+class CustomQuerySet(QuerySetNoCache):
+    """
+    All custom QuerySet classes should inherit from this class rather than QuerySet
+    or QuerySetNoCache. There are two reasons for this. Firstly, all QuerySets should
+    not cache their query results by default, so QuerySetNoCache must be inherited.
+    Secondly, QuerySetNoCache's implementation of the `cache` instance method
+    explicitly returns an instance of QuerySet which effectively removes any
+    customized behavior for that QuerySet.
+    """
+
+    def cache(self):
+        """
+        Currently, there is no reason to have a caching version of any of our
+        custom QuerySet classes. If that ever changes, this method should be
+        implemented to return a caching version of the custom QuerySet. This
+        implementation exists in order to avoid bugs resulting from the
+        unexpected default behavior of `cache`.
+
+        As of mongoengine 0.10, the default implementation of ``cache`` cannot
+        be used as it explicitly creates a QuerySet object, thus removing any
+        custom behavior of the custom QuerySet.
+        """
+        raise NotImplementedError()
+
+    def __len__(self):
```

Comment:
I'm honestly fine with any of those; the main thing is that the name be a little more descriptive than `CustomQuerySet`, but not so descriptive that we have to rename it to add a method without feeling bad about having to refactor it later.

That's where `QuerySetPreventCache` came from...not only does it not cache, it goes so far as preventing you from converting it to a caching QuerySet by exploding when you call `.cache()`. But like I said, all the names are probably fine, and far less important than the docstrings, which are great. Your call.